callout/src/main/java/com/google/apigee/ProtoMessageBuilders/ExecutionContextProtoMessageBuilder.java

```
@@ -0,0 +1,142 @@
+   *
+   * @param fault {@link Fault} used to construct the Protocol Buffer Message.
+   * @return Fault Protocol Buffer Message
+   */
+  private ExecutionOuterClass.ExecutionContext.Fault buildFault(Fault fault) {
+    return ExecutionOuterClass.ExecutionContext.Fault.newBuilder()
+        .setCategory(getFaultCategory(fault))
+        .setSubCategory(fault.getSubCategory())
+        .setName(fault.getName())
+        .setReason(fault.getReason())
+        .putAllAttributes(buildAttributesMap(fault))
+        .build();
+  }
+
+  /**
+   * Determines the appropriate Category for the Fault Protocol Buffer Message
+   *
+   * @param fault Fault from which to extract the Category
+   * @return Fault.Category Protocol Buffer enumerator value
+   */
+  private ExecutionOuterClass.ExecutionContext.Fault.Category getFaultCategory(Fault fault) {
+    switch (fault.getCategory()) {
+      case Messaging:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.MESSAGING;
+      case Step:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.STEP;
+      case Transport:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.TRANSPORT;
+      case System:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.SYSTEM;
+      default:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.UNKNOWN_CATEGORY;
+    }
+  }
+
+  /**
+   * Constructs the String to Any attributes map for a Fault protocol buffer message
+   *
+   * @param fault Fault from which attributes map is extracted from
+   * @return Map of String to Any for protocol buffer message
+   */
+  private Map<String, Any> buildAttributesMap(Fault fault) {
+    Map<String, Any> attributesMap = new HashMap<>();
+    fault
+        .getAttributes()
+        .forEach(
+            (key, value) -> {
+              try {
+                attributesMap.put(key, Any.parseFrom(objectToByteArray(value)));
+              } catch (Exception exception) {
```

This catch-block effectively silences the exception.  Are there any downsides to that?

---

changed this to throw an uncheckedioexception